### DISCUSSION TITLE
set date and time - scripting

### DISCUSSION THREAD

**Discussion by fzs1933** [Stellarium/stellarium#2070] [Q&A] [answered]
Hi,
again

As I wrote once **I don't know javascript** therefore use only stellarium commands even it is too long.
I want to show the starry nights at North Pole and the Equator - night to night. (this is a part of a bigger project but very important)

Before starts the "equator dates and times", time change to "now" date and time for a second. I tried it with 0.21 and 0.21.2. Same.
And tried a lot.
In fact, when I just try without north pole dates, stellarium does the same.

I am afraid, it wasn't too comprehensible...
Here is the code: [https://github.com/fzs1933/seasons/blob/main/night_to_night](url)
Thanks for any answer (and your time)
Zsolt

**Comment by axd1967** [comment]
Hello @fzs1933 

I tried your script, it runs fine at first sight? What is the problem? 

The scripts starts at the North Pole and ends at the Equator?

The only thing that I noticed is that it does not have a `.ssc` filename extension, this could make your life a bit more difficult.

Congratulations for posting your script on GitHub, because that way we can best help you! If you think there is a problem IN your script, consider opening an [issue in your repository](https://github.com/fzs1933/seasons/issues) and we will help you there ?

**Reply by fzs1933** [reply]
Hi, @axd1967 

Thanks for your answer.
I have found that when we go over to the equator (2022-03-21) and setting date and time, it displays for a moment the sky corresponding to the current time, not the one defined in the script (2022-03-21). And then does it as I defined but this "bug" is annoying...

It's possible that I don't see something small mistake.
I write scripts in Visual Studio Code as .js and save as .ssc.

**Reply by fzs1933** [reply]
addition:
no, didn't run correctly at first

**Comment by axd1967** [comment, marked as answer]
yes, you are right!

just before continuing from equator, there is indeed a time jump.

the bug is that the `setDate()` time formats are wrong: you need to use "2021-03-20T0x:00:00" rather than "2021-03-20Tx:0:00".


remember to check the MAIN log, you will find the error there (and unfortunately, not in the script editor log, which IS a bug).
![image](https://user-images.githubusercontent.com/3529789/143782516-90c911fc-9efe-4b6d-81b3-0be6a3dbcf95.png)

(also, is there a need to re-specify the landscape or `moveToAltAzi()` or `zoomTo()` ?)

**Reply by fzs1933** [reply]
It was front of me. Oh, what a .......... :)
So, two digits are required.
Thanks a lot!

**Reply by fzs1933** [reply]
no, no need re-specify but that line just copied and stayed that way

**Reply by axd1967** [reply]
aaa :-) you know, **copy-paste is evil** :-)

**Reply by fzs1933** [reply]
Finally
I tried and of course it works perfectly.
Daylight saving wasn't simple but it is correct too.
Thanks again
Regards
Zsolt
